sdk/core/azure-core/src/datetime.cpp

```diff
@@ -774,9 +779,84 @@ DateTime DateTime::FromString(std::string const& dateString, DateFormat format)
   }
   else
   {
-    throw DateTimeException("Unrecognized date format.");
+    throw std::invalid_argument("Unrecognized date format.");
   }
 
-  result.m_interval = static_cast<IntervalType>(secondsSince1601 * TicksPerSecond + fracSec);
-  return result;
+  return DateTime(std::chrono::seconds(secondsSince1601) + DateTime::Duration(fracSec));
+}
+
+DateTime::DateTime(
+    int16_t year,
+    int8_t month,
+    int8_t day,
+    int8_t hour,
+    int8_t minute,
+    int8_t second)
+{
+  // We should combine creation/validation logic, so it is reusable from both parsing functions and
+  // from this constructor
+  if (year > MaxYear || year < MinYear)
+  {
+    throw std::invalid_argument(
+        year > MaxYear ? "The requested year exceeds the year 9999."
+                       : "The requested year is less than the year 1601.");
+  }
+
+  if (month <= 0 || month > 12)
+  {
+    throw std::invalid_argument("Invalid month value.");
+  }
+
+  if (day <= 0 || day > 31)
+  {
+    throw std::invalid_argument("Invalid day value.");
+  }
+
+  if (hour < 0 || hour > 23)
+  {
+    throw std::invalid_argument("Invalid hour value.");
+  }
+
+  if (minute < 0 || minute > 60)
+  {
+    throw std::invalid_argument("Invalid minute value.");
+  }
+
+  if (second < 0 || second > 60)
```

Comment:
Is 60 valid?

Reply:
60 is allowed, it is leap seconds, parser would accept it. But not for minutes, good catch.